test/testutil/stanza.c

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2017 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+#include <assert.h>
+#include <errno.h>
+#include <stdio.h>
+#include <string.h>
+#include <ctype.h>
+
+#include "e_os.h"
+#include "../testutil.h"
+
+int test_start_file(STANZA *s, const char *testfile)
+{
+    TEST_info("Reading %s", testfile);
+    set_test_title(testfile);
+    memset(s, 0, sizeof(*s));
+    if (!TEST_ptr(s->fp = BIO_new_file(testfile, "rb")))
+        return 0;
+    s->test_file = testfile;
+    return 1;
+}
+
+int test_end_file(STANZA *s)
+{
+    TEST_info("Completed %d tests with %d errors and %d skipped",
+              s->numtests, s->errors, s->numskip);
+    BIO_free(s->fp);
+    return 1;
+}
+
+/*
+ * Read a PEM block.  Return 1 if okay, 0 on error.
+ */
+static int read_key(STANZA *s)
+{
+    char tmpbuf[128];
+
+    if (s->key == NULL) {
+        if (!TEST_ptr(s->key = BIO_new(BIO_s_mem())))
```

Comment:
Yeah, I know.  I tended to put all "infrastructure" things within a ```TEST_``` macro because of the convenient debug and location stuff even though it *is not really* a test.  Maybe an assert? :)